pkg/cmd/ssh/ssh_patch_options.go

```diff
@@ -0,0 +1,343 @@
+
+	if err != nil {
+		return err
+	} else if !hasChanges {
+		return nil
+	}
+
+	return gardenClient.PatchBastion(ctx, bastion, oldBastion)
+}
+
+func getGardenClient(f util.Factory) (target.Manager, target.Target, gardenClient.Client, error) {
+	manager, err := f.Manager()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// sshTarget is the target used for the run method
+	sshTarget, err := manager.CurrentTarget()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	gardenClient, err := manager.GardenClient(sshTarget.GardenName())
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return manager, sshTarget, gardenClient, nil
+}
+
+func (o *SSHPatchOptions) Run(f util.Factory) error {
+	_, sshTarget, gardenClient, err := getGardenClient(f)
+	if err != nil {
+		return err
+	}
+
+	// TODO: handle e.g. interrupts or is timeout enough?
+	ctx, cancel := context.WithTimeout(f.Context(), 30*time.Second)
+	defer cancel()
+
+	err = patchBastion(ctx, o, gardenClient, sshTarget)
+
+	return err
+}
+
+func (o *SSHPatchOptions) Complete(f util.Factory, cmd *cobra.Command, args []string) error {
+	manager, _, gardenClient, err := getGardenClient(f)
+	if err != nil {
+		return err
+	}
+
+	if err := o.sshBaseOptions.Complete(f, cmd, args); err != nil {
+		return err
+	}
+
+	authInfo, err := getAuthInfo(f.Context(), manager)
+	if err != nil {
+		return err
+	}
+
+	if o.BastionName == "" {
+		bastionNames, err := getBastionNamesOfUser(f.Context(), gardenClient, authInfo)
+		if err != nil {
+			return err
+		}
+
+		if len(bastionNames) == 1 {
+			fmt.Fprintf(o.IOStreams.Out, "Auto-selected bastion '%s'\n", bastionNames[0])
+			o.BastionName = bastionNames[0]
+		} else {
+			fmt.Fprintf(o.IOStreams.Out, "Could not auto-select bastion because there are none/multiple. Specify the bastion with --bastion-name.\n")
+		}
+	}
+
+	return nil
+}
+
+func (o *SSHPatchOptions) getBastionNameCompletions(f util.Factory, cmd *cobra.Command, args []string) ([]string, error) {
+	completions := []string{}
+
+	manager, _, gardenClient, err := getGardenClient(f)
+	if err != nil {
+		return completions, err
+	}
+
+	// TODO: handle interrupts of some sort?
+	ctx, cancel := context.WithTimeout(f.Context(), 30*time.Second)
+	defer cancel()
+
+	authInfo, err := getAuthInfo(ctx, manager)
+	if err != nil {
+		return completions, err
+	}
+
+	bastions, err := getBastionsOfUser(ctx, gardenClient, authInfo)
+	if err != nil {
+		return completions, err
+	}
+
+	for _, bastion := range bastions {
+		completion := fmt.Sprintf("%s\t created at %s", bastion.Name, bastion.CreationTimestamp.Format(time.RFC3339))
+		completions = append(completions, completion)
+	}
+
+	sort.Strings(completions)
+
+	return completions, nil
+}
+
+func (o *SSHPatchOptions) Validate() error {
+	if err := o.sshBaseOptions.Validate(); err != nil {
+		return err
+	}
+
+	if o.BastionName == "" {
+		return fmt.Errorf("bastion name flag not provided and unable to automatically determine bastion name")
+	}
+
+	return nil
```

Comment:
shouldn't it be also validated that there are cidrs set on the options? Otherwise this command wouldn't do anything

Reply:
@petersutter The CIDRs and the validation logic was moved to the "base options". So CIDRs should be checked through `o.sshBaseOptions.Validate()`. Is that approach ok for you?

Reply:
✅